File: leetcode/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/solution.kt
class Solution {
    fun isPrefixOfWord(sentence: String, searchWord: String): Int {
        val i = sentence
            .splitToSequence(' ')
            .indexOfFirst { it.startsWith(searchWord) }
        if(i >= 0) {
            return i+1
        }
        return i
    }
}

File: leetcode/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/solution.rs
impl Solution {
    pub fn is_prefix_of_word(sentence: String, search_word: String) -> i32 {
        sentence.split_whitespace()
            .position(|s| s.starts_with(&search_word))
            .map_or(-1, |i| 1+i as i32)
    }
}
